modules/cockroachdb/cockroachdb.go

```diff
@@ -0,0 +1,254 @@
+		return "", err
+	}
+
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return connString(c.opts, host, port), nil
+}
+
+// TLSConfig returns config necessary to connect to CockroachDB over TLS.
+func (c *CockroachDBContainer) ConnectionTLS() (*tls.Config, error) {
+	return connTLS(c.opts)
+}
+
+// RunContainer creates an instance of the CockroachDB container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*CockroachDBContainer, error) {
+	o := defaultOptions()
+	req := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image: defaultImage,
+			ExposedPorts: []string{
+				defaultSQLPort,
+				defaultAdminPort,
+			},
+		},
+	}
+
+	// apply options
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			apply(&o)
+		}
+		opt.Customize(&req)
+	}
+
+	addEnvs(&req, o)
+	if err := addCmd(&req, o); err != nil {
+		return nil, err
+	}
+	if err := addWaitingFor(&req, o); err != nil {
+		return nil, err
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// copy TLS files
+	if err := addTLS(ctx, container, o); err != nil {
+		return nil, err
+	}
+
+	// start
+	if err := container.Start(ctx); err != nil {
+		return nil, err
+	}
+	return &CockroachDBContainer{Container: container, opts: o}, nil
+}
+
+func addCmd(req *testcontainers.GenericContainerRequest, opts options) error {
+	req.Cmd = []string{
+		"start-single-node",
+		"--store=type=mem,size=" + opts.StoreSize,
+	}
+
+	// authN
+	if opts.TLS != nil {
+		if opts.User != defaultUser {
+			return fmt.Errorf("unsupported user %s with TLS, use %s", opts.User, defaultUser)
+		}
+		if opts.Password != "" {
+			return fmt.Errorf("cannot use password authentication with TLS")
+		}
+	}
+
+	switch {
+	case opts.TLS != nil:
+		req.Cmd = append(req.Cmd, "--certs-dir="+certsDir)
+	case opts.Password != "":
+		req.Cmd = append(req.Cmd, "--accept-sql-without-tls")
+	default:
+		req.Cmd = append(req.Cmd, "--insecure")
+	}
+
+	return nil
+}
+
+func addEnvs(req *testcontainers.GenericContainerRequest, opts options) {
+	if req.Env == nil {
+		req.Env = make(map[string]string)
+	}
+
+	req.Env["COCKROACH_DATABASE"] = opts.Database
+	req.Env["COCKROACH_USER"] = opts.User
+	req.Env["COCKROACH_PASSWORD"] = opts.Password
+}
+
+func addWaitingFor(req *testcontainers.GenericContainerRequest, opts options) error {
+	var tlsConfig *tls.Config
+	if opts.TLS != nil {
+		cfg, err := connTLS(opts)
+		if err != nil {
+			return err
+		}
+		tlsConfig = cfg
+	}
+
+	req.WaitingFor = wait.ForAll(
+		wait.ForHTTP("/health").WithPort(defaultAdminPort),
+		wait.ForSQL(defaultSQLPort, "pgx/v5", func(host string, port nat.Port) string {
+			connStr := connString(opts, host, port)
+			if tlsConfig == nil {
+				return connStr
+			}
+
+			// register TLS config with pgx driver
+			connCfg, err := pgx.ParseConfig(connStr)
+			if err != nil {
+				panic(err)
```

Comment:
I wasn't sure how best to go about this as we need the host & port to build the connection string before adding TLS config.

One option is creating a custom `wait.Strategy` but that felt heavy-handed. Thoughts?

Reply:
We could explicitly mention the panic in the docs, in the TLS section.

Reply:
Although not included in other modules, I'd like to star having this kind of mentions in the docs. Could you please add that to the docs?